types/d3-array/index.d.ts

```diff
@@ -311,29 +314,64 @@ export function zip<T>(...arrays: Array<ArrayLike<T>>): T[][];
 // Histogram
 // --------------------------------------------------------------------------------------
 
-export interface Bin<Datum, Value extends number | Date> extends Array<Datum> {
-    x0: Value;
-    x1: Value;
+export interface Bin<Datum, Value extends number | Date | undefined> extends Array<Datum> {
+    x0: Value | undefined;
+    x1: Value | undefined;
 }
 
 /**
  * Type definition for threshold generator which returns the count of recommended thresholds
  */
-export type ThresholdCountGenerator = (values: ArrayLike<number>, min?: number, max?: number) => number;
+export type ThresholdCountGenerator<Value extends number | undefined> = (values: ArrayLike<Value>, min?: number, max?: number) => number;
```

Comment:
This will not be backward compatibility for people working in strict function type. The exact equivalent with previous code is `= number>`. Should we care?

Reply:
On this one, I think it is fair game to change to `<Value extends number | undefined = number | undefined>` and add the default including `undefined`.

Adding generics without a default is definitely backwards breaking, see the necessary changes to *d3-contour* you already had to make. These would not be required.

W.r.t. backwards breaking under `strictFunctionTypes`:

While unfortunate that we are still working through the technical debt of all the validations for `strictNullChecks` and `strictFunctionTypes` (albeit at a much faster clip with your help), we always knew that the added strictness may be breaking folks who used `strictXYZ` already with the unvalidated definitions. That was also the clear reason not to update the `tsconfig` and portrait them as `strictXYZ` before having done the actual work.